packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,3661 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/widgets.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'ink_well.dart';
+import 'material.dart';
+
+/// Begin a Material 3 ink sparkle ripple, centered at the tap or click position
+/// relative to the [referenceBox].
+///
+/// This effect relies on a shader, and therefore hardware acceleration.
+/// Currently, this is only supported by C++ engine platforms: Android, iOS,
+/// MacOS, Windows, and Linux, etc. It can run on other platforms with emulation
+/// in software, but will not be performant. Hardware enabled support for
+/// CanvasKit web can be tracked here:
+///  - https://github.com/flutter/flutter/issues/85238
+///
+/// To use this effect, pass an instance of [splashFactory] to the
+/// `splashFactory` parameter of either the Material [ThemeData] or any
+/// component that has a `splashFactory` paramter, such as buttons:
+///  - [ElevatedButton]
+///  - [TextButton]
+///  - [OutlinedButton]
+///
+/// The [controller] argument is typically obtained via
+/// `Material.of(context)`.
+///
+/// If [containedInkWell] is true, then the ripple will be sized to fit
+/// the well rectangle, then clipped to it when drawn. The well
+/// rectangle is the box returned by [rectCallback], if provided, or
+/// otherwise is the bounds of the [referenceBox].
+///
+/// If [containedInkWell] is false, then [rectCallback] should be null.
+/// The ink ripple is clipped only to the edges of the [Material].
+/// This is the default.
+///
+/// When the ripple is removed, [onRemoved] will be called.
```

Comment:
Might want to add a "See also" pointing to the Material 3 spec for this. Also a usage example would go nice here.

Reply:
Is there a material 3 spec for the ripple at material.io?

Reply:
Sorry, I just assumed it would be there. I don't see one.